File: Medium/11. Container With Most Water/Solution.c
int maxArea(int* height, int heightSize){
    int m = 0, start = 0, end = heightSize - 1, minHeight, res;
    while (start != end){
        minHeight = height[start];
        if (height[end] < minHeight)
            minHeight = height[end];
        res = minHeight * (end - start);
        if (res > m)
            m = res;
        if (height[start] < height[end]) {
            start++;
        } else {
            end--;
        } 
    }
    return m;
}

File: Medium/11. Container With Most Water/Solution.py
class Solution(object):
    def maxArea(self, height):
        start = 0
        end = len(height) - 1
        m = 0
        while start < end:
            res = min(height[start], height[end]) * (end - start)
            if res > m:
                m = res
            if height[start] < height[end] :
                start += 1
            else:
                end -= 1
        return m